test/lib/mutation_source_test.cc

```diff
@@ -1773,6 +1773,38 @@ void run_mutation_source_tests(populate_fn_ex populate, bool with_partition_rang
                     ms.make_reader(s, std::move(permit), pr, slice, pc, std::move(tr), fwd, mr_fwd));
         });
     }, with_partition_range_forwarding);
+
+    std::vector<mutation> reversed_mutations;
+    std::list<query::partition_slice> reversed_slices;
+
+    // read in reverse
+    run_mutation_reader_tests([populate, &reversed_mutations, &reversed_slices] (schema_ptr s, const std::vector<mutation>& m, gc_clock::time_point t) -> mutation_source {
+        auto table_schema = s->make_reversed();
+
+        reversed_mutations.clear();
+        reversed_mutations.reserve(m.size());
+        for (const auto& mut : m) {
+            reversed_mutations.emplace_back(reverse(mut));
+        }
+
+        reversed_slices.clear();
```

Comment:
It would be better to not modify shared state in case we want to parallelize the tests in the future.

Reply:
I don't know how to avoid it. I have to store this state somewhere, otherwise I get use-after-free.

Reply:
You could allocate it in a shared_ptr and carry around. There is now chained_delegating_reader which allows you to install a lambda which lives as long as the reader lives.

If you think it's too much trouble, we can defer this.

Reply:
It will get easier when we switch to native reverse slice.

Reply:
This lambda should live as long as the tests are running, no?
Can't `reversed_mutations` and `reversed_slices` at least become local variables of this lambda? You won't need to `clear` them.

Reply:
I'll see if I can reduce the scope of these states.